cni-plugin/integration/testutil/test_util.go

```diff
@@ -39,6 +40,70 @@ func (r *TestRunner) walkConfDir() (map[string]struct{}, error) {
 	return fileNames, nil
 }
 
+// Checks that the embedded linkerd config is of the expected form
+// and contains the right values:
+//
+//	"linkerd": {
+//	   "incoming-proxy-port": 4143,
+//	   "outgoing-proxy-port": 4140,
+//	   "proxy-uid": 2102,
+//	   "ports-to-redirect": [],
+//	   "inbound-ports-to-ignore": ["4191","4190"],
+//	   "simulate": false,
+//	   "use-wait-flag": false
+//	 }
+func checkLinkerdCniConf(wrapperConf map[string]any) error {
```

Comment:
Not a blocker but should we create a wrapper struct to more easily deserialize this config in a follow-up PR? Do you think that'd make things a bit easier to work with? Something similar to https://github.com/linkerd/linkerd2-proxy-init/blob/main/cni-plugin/main.go#L44-L55. Perhaps we could even move that in a shared (internal) library 🤷🏻 

This test is good to assert default values are fine, but we might want to try with some overrides in the future?

Reply:
I think using the struct makes sense but I can't figure out how to import `ProxyInit` from `main.go` into this test package.

Reply:
I switched to using `mapstructure` which has more options for dynamically decoding structs, take a look.